### PR TITLE
[consensus] integrate onchain_config 

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

Bridge the move resource with corresponding on-chain config Rust struct, so it can be used in consensus.

This PR adds the DiemConsensusConfig::initialize in genesis and add the config id in REGISTRY, so existing chain has to initialize the config before deploying this change otherwise the code would panic with failing to access the on chain config.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

Added test.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/diem/diem/tree/main/developers.diem.com, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

File: types/src/on_chain_config/consensus_config.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+use serde::{Deserialize, Serialize};
+
+/// The on-chain consensus config, in order to be able to add fields, we use enum to wrap the actual struct.
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub enum OnChainConsensusConfig {
+    V1(ConsensusConfigV1),
+}
+
+impl OnChainConsensusConfig {
+    #[allow(dead_code)]
```
Comment (gregnazario) [inline types/src/on_chain_config/consensus_config.rs:12]: Is this still not used anywhere yet?

**Review by gregnazario — DISMISSED** [review]
I got you're pretty much replacing `Script` with `TransactionPayload` and adding the `OnChainConfig` for consensus.  Looks good to me.

File: types/src/on_chain_config/consensus_config.rs (hunk 1 above)
Comment (zekun000) [reply types/src/on_chain_config/consensus_config.rs:12]: not yet, hopefully next PR

**Review by dmitri-perelman — DISMISSED** [review]
Looks good!

**Review by gregnazario — DISMISSED** [review]

**Pull request closed without merging**